addon/components/frost-date-picker.js

```diff
@@ -40,6 +47,24 @@ export default Component.extend(EventsProxyMixin, {
 
   // == Computed Properties ===================================================
 
+  @readOnly
+  @computed('value')
+  _validatedValue (value) {
+    if (value) {
+      return moment(value, this.get('format'))
```

Comment:
I feel like this cp should be doing the validation with `isValid` and returning the formatted date and return false otherwise. Everywhere else, you're repeating that logic.